---
Title: "TranscriptomicProfilingTCGA"
author: "Michael Skaro PhD"
Date of work begin: "2023-11-29"
Data output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
## Transcriptomic profiling of TCGA database




The objective of this study is to profile The Cancer Genomes Atlas for expression
data in tumor and normal samples. The expression data will be analyzed with the 
intent to illuminate gene expression differences in genes encoding surface proteins.

Steps in the analysis:

- The TCGAbiolinks package will be used to access and analyze the expression data. 

- The differentially expressed genes for each of the indication will be listed.

- Upregualted genes in each indication will be intersected with the list of genes
encoding surface proteins.

- The 25th, 50th, 75th and 100th percentile of each of the GESP gene's expression 
will be listed for each indication in the TCGA database 
(TPM/FPKM availability dependent).

Visualizations:

- Heatmap of FC/Normal in indications where matched normal is available.
- Venn Diagram of intersection from GESP list and DEG list for each indication.
- Box plot for each differentially expressed gene in the intersected list. 
- PCA plot for tumor and normal tissues based solely on the DE GESPs

Filtering for Top30

A list of the top30 genes in each indication will be conducted. The genes
will be ranked on their fold change over normal and a pvalue below 0.01. 
Then the genes will be filtered on criteria described by ARTbio team. 

Filtering criteria:

- Exclude targets with tumor enrichment but ‘significant’ expression in colon, 
small intestine or haem compartments. The ARTbio team has not defined significant
but there is literature precedent to suggest high expression of surface targets
that have been successfully targeted with an ADC it 35TPM. We can use that as a 
minimum. 

- The average expression of constitutively expressed genes such as GAPDH
will be used as the upper limit. The median of expression in the colon between the
two will be used as a level for flagging. 

Non-visual outputs:

- Correlation between expression in TCGA and in the HPA database may/may not be
possible depending on the availability of data. Case by case basis will be used
and will be flagged as NA if this portion of the analysis cannot be executed 
due to missingness of the data. 


```{r libraries, echo=FALSE, message=FALSE}

# The libraries that listed below are required to reproduce the analysis. 
# BiocManager::install("TCGAbiolinks")
library(BiocManager)
library(TCGAbiolinks)
library(dplyr)
library(tidyverse)
library(ggplot2)
library(stringr)
library(EDASeq)
library(edgeR)
library(org.Hs.eg.db)
library(UCSCXenaTools)
library(R.utils)
library(circlize)
library(ComplexHeatmap)
library(fgsea)
```




#### Differential expression analysis in TCGA

This function will satisfy the first objective in that it will process and analyze
each of the projects in the TCGA database and produce a results table. The results
table will be used to 


```{r Function for sourcing expression data, echo=FALSE}

differential_expression_data_TCGA_projects <- function(proj){
  # Input: String, TCGA project 4 letter code
  # Output: expression data from TCGA project passed, type: S4
  # Libraries used: TCGAbiolinks
  
  query_exp <- GDCquery(
  project = str_glue("TCGA-{proj}"),
  data.category = "Transcriptome Profiling",
  data.type = "Gene Expression Quantification", 
  workflow.type = "STAR - Counts")
  
  print("query exp done")
  GDCdownload(query_exp)
  Rnaseq.SE <- GDCprepare(query_exp)
  RNAseq_CorOutliers <- TCGAanalyze_Preprocessing(Rnaseq.SE)
  dataNorm <- TCGAanalyze_Normalization(tabDF = RNAseq_CorOutliers, geneInfo =  geneInfoHT)
  dataFilt <- TCGAanalyze_Filtering(tabDF = dataNorm,method = "quantile", qnt.cut =  0.25)
  samplesNT <- TCGAquery_SampleTypes(barcode = colnames(dataFilt),typesample = c("NT"))
  samplesTP <- TCGAquery_SampleTypes(barcode = colnames(dataFilt), typesample = c("TP"))
  
  # take the length of the samplesNT and the length of the samplesTP, if the length of the
  # samples is not >10 move on. 
  
  if(length(samplesNT) <10){
    
    print(str_glue("{proj} does not have enough samples, skipping this project"))

    dataDEGs <- NA    
    
  } else{

  dataDEGs <- TCGAanalyze_DEA(
    mat1 = dataFilt[,samplesNT],
    mat2 = dataFilt[,samplesTP],
    Cond1type = "Normal",
    Cond2type = "Tumor",
    #fdr.cut = 0.01,
    #logFC.cut = 1,
    method = "glmLRT")
  
  save(dataDEGs,file = str_glue("Res_TCGA_{proj}.Rdata"))
  
  data.table::fwrite(file = str_glue("Res_Gene_expression_TCGA_{proj}.txt"), x = dataDEGs)
  
  
  }
  
  return(dataDEGs)
  
}

```

```{r Create function for labeling results, echo=FALSE, message=FALSE}

label_results_table <- function(dataDEGs, proj, GESPs){
  dat <- dataDEGs %>%
    tibble::rownames_to_column("ENSEMBL_ID") %>%
    dplyr::mutate(TGCA_project = proj) %>%
    dplyr::left_join(GESPs, by = c("gene_name", "ENSEMBL_ID")) %>%
    dplyr::filter(!is.na(surface))
  dat$diff <- "notsig"
  dat$diff[dat$logFC> 1 & dat$FDR<0.01] <- "up_regulated"
  dat$diff[dat$logFC< -1 & dat$FDR<0.01] <- "down_regulated"
  
  ensgs <- dat %>%
    dplyr::pull(ENSEMBL_ID)
  
  cols <- c("SYMBOL","ENZYME", "GENENAME")
  val <- as.data.frame(select(org.Hs.eg.db, keys=ensgs, columns=cols, keytype="ENSEMBL"))
  colnames(val) <- c("ENSEMBL_ID", "gene_name", "Enzyme", "Gene_class")
  
  dat <- dat %>%
    dplyr::left_join(val, by = c("ENSEMBL_ID", "gene_name")) %>%
    dplyr::arrange(desc(logFC))
  
  return(dat)
  
}


```


```{r Download XenaData for gene expression TPMs}

data(XenaData)

geneExpression <- XenaData %>% dplyr::filter(DataSubtype == "gene expression RNAseq")
percentileRank <- geneExpression %>% dplyr::filter(Unit =="percentile rank")

# pancancer TPMs

# tcga_RSEM_gene_tpm, toilHub
xo <- XenaGenerate(subset = XenaHostNames=="toilHub") %>% 
  XenaFilter(filterDatasets = "tcga_RSEM_gene_tpm")

xe <- XenaQuery(xo)
xd <- XenaDownload(xe, destdir = "/Volumes/My Passport/artBio_consultation/data/", download_probeMap = TRUE)

# GtexPanNormal

# gtex_RSEM_gene_tpm, toilHub


xo <- XenaGenerate(subset = XenaHostNames=="toilHub") %>% 
  XenaFilter(filterCohorts = "GTEX", ignore.case = FALSE) %>%
  XenaFilter(filterDatasets = "gtex_RSEM_gene_tpm", ignore.case = FALSE)
xe <- XenaQuery(xo)
xd <- XenaDownload(xe, destdir = "/Volumes/My Passport/artBio_consultation/data/", download_probeMap = TRUE)

# download the data for the percentiles
list.of.datasets <- unique(percentileRank$XenaDatasets)
list.of.cohorts <- unique(percentileRank$XenaCohorts)
for(i in 1:length(list.of.datasets)){
  dat <- list.of.datasets[i]
  cohort <- list.of.cohorts[i]
  #host.name <- "tcgaHub"
  xo = XenaGenerate(subset = XenaHostNames == "tcgaHub") %>%
    XenaFilter(filterDatasets = dat, ignore.case = FALSE)
  xe <- XenaQuery(xo)
  xd <- XenaDownload(xe, destdir = "/Volumes/My Passport/artBio_consultation/data/gene_expression_ranks/", download_probeMap = TRUE)

  
}

```

```{r Build function for visualizing barplot }

DGE.bar.plot <- function(val){
  res.table <- data.table::fread(val)
  res.table$diff <- "notsig"
  res.table$diff[res.table$logFC> 1 & res.table$FDR<0.01] <- "up_regulated"
  res.table$diff[res.table$logFC< -1 & res.table$FDR<0.01] <- "down_regulated"
  proj <- str_split(val, pattern = "_")
  proj <- proj[[1]][6]
  proj <- str_split(proj, pattern = "\\.")
  proj <- proj[[1]][1]
  
  res.table <- res.table %>%
    dplyr::group_by(diff) %>%
    dplyr::summarise(Count = n()) %>%
    dplyr::mutate(diff = str_replace_all(string = diff, pattern = "down_regulated", replacement = "DownRegulated")) %>%
    dplyr::mutate(diff = str_replace_all(string = diff, pattern = "up_regulated", replacement = "UpRegulated")) %>%
    dplyr::mutate(diff = str_replace_all(string = diff, pattern = "notsig", replacement = "NotSignificant")) 
  res.table$diff <- factor(res.table$diff, levels = c("DownRegulated", "UpRegulated", "NotSignificant"))
  colors <- c("DownRegulated" = "Blue", "UpRegulated" = "firebrick", "NotSignificant" = "gray52")
  p <- ggplot2::ggplot(data = res.table, aes(x = diff, y = Count, fill = diff)) +
    geom_bar(stat = "identity", position = "dodge") +
    scale_fill_manual(values = colors) +
    geom_text(aes(label = Count), vjust = -.2) +
    labs(fill = "DE Genes", caption = "absL2FC >1, adj.pvalue <0.01") +
    ggtitle(str_glue("Differentially Expressed Genes in TCGA-{proj}")) +
    xlab("Differential Expression results vs. adjacent Normal") +
    theme_bw() +
    theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))
    
  return(p)
}

GESP.bar.plot <- function(val){
  res.table <- data.table::fread(val) %>%
    dplyr::group_by(diff) %>%
    dplyr::summarise(Count = n()) %>%
    dplyr::mutate(diff = str_replace_all(string = diff, pattern = "down_regulated", replacement = "DownRegulated")) %>%
    dplyr::mutate(diff = str_replace_all(string = diff, pattern = "up_regulated", replacement = "UpRegulated")) %>%
    dplyr::mutate(diff = str_replace_all(string = diff, pattern = "notsig", replacement = "NotSignificant")) 
  res.table$diff <- factor(res.table$diff, levels = c("DownRegulated", "UpRegulated", "NotSignificant"))
  colors <- c("DownRegulated" = "Blue", "UpRegulated" = "firebrick", "NotSignificant" = "gray52")
  p <- ggplot2::ggplot(data = res.table, aes(x = diff, y = Count, fill = diff)) +
    geom_bar(stat = "identity", position = "dodge") +
    scale_fill_manual(values = colors) +
    geom_text(aes(label = Count), vjust = -.2) +
    labs(fill = "DE Genes") +
    ggtitle(str_glue("Differentially Expressed GESPs in TCGA-{proj}")) +
    theme_bw() +
    theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))
    
  return(p)
}


```

```{r Pull DE genes from differential expression analsyis}

# The differentially expressed genes from a res.table 
gene.list.from.res.tables <- function(res.table){
  
  gene.list <- res.table %>%
    dplyr::filter(diff != "notsig") %>%
    dplyr::pull(ENSEMBL_ID)
  
  return(gene.list)
  
}


```


```{r load gene expression data from a project}
setwd("/Volumes/My Passport/artBio_consultation/")
val <- "/Volumes/My Passport/artBio_consultation/data/tcga_RSEM_gene_tpm.gz"
tcga.tmps <- data.table::fread(val)
tcga.tmps <- tcga.tmps %>%
  dplyr::rename("ENSEMBL" = "sample") %>%
  dplyr::mutate(ENSEMBL = stringr::str_sub(string = ENSEMBL, start = 1, end = 15))
tcga.clin.metadata <- data.table::fread("/Volumes/My Passport/artBio_consultation/data/TCGA_patient_metadata.txt")
de.gesps <- data.table::fread("/Volumes/My Passport/artBio_consultation/results/DE_gesps.txt") %>%
  dplyr::rename("ENSEMBL" = "DE_GESPs") %>%
  dplyr::pull()
# take large matrix, distill down to TCGA project of interest


# iterate over the 16 projects for the HMs, select only the GESPs in the analysis
# read in the results table
# select the genes add to a running temp
counter = 0
list.of.files <- list.files("/Volumes/My Passport/artBio_consultation/results/Rdata")
fil <- list.of.files[1]
for(i in 1:length(list.of.files)){
  fil <- list.of.files[i]
  proj <- str_split(fil, pattern = "_")
  proj <- proj[[1]][3]
  proj <- str_split(proj, pattern = "\\.")
  proj <- proj[[1]][1]
  load(str_glue("/Volumes/My Passport/artBio_consultation/results/Rdata/{fil}"))
  
  res <- dataDEGs %>%
    tibble::rownames_to_column("ENSEMBL") %>%
    dplyr::filter(ENSEMBL %in% de.gesps) %>%
    dplyr::mutate(Project = proj) 
  
  if(counter ==0){
    temp <- res
    counter = counter +1
  } else {
    temp <- rbind(temp, res)
  }
  
  
}
dat <- temp
rm(res,temp)

dat$diff <- "notsig"
dat$diff[dat$logFC> 1 & dat$FDR<0.01] <- "up_regulated"
dat$diff[dat$logFC< -1 & dat$FDR<0.01] <- "down_regulated"

# Group by the Project and the gene, if the gene as not expressed in the project, pivot longer and fill with 0/NA
# Pivot wider for the expression matrix
# cluster the genes
# create a heatmap for the sruface proteins
# annoatate the proteins
# conudct gene set enrichment analysis

dat.long <- dat %>%
  dplyr::mutate(diff = str_replace_all(string = diff, pattern = "up_regulated", replacement = "Up-Regulated")) %>%
  dplyr::mutate(diff = str_replace_all(string = diff, pattern = "down_regulated", replacement = "Down-Regulated")) %>%
  dplyr::select(Project, gene_name, logFC) %>%
  tidyr::pivot_wider(id_cols = Project, names_from = gene_name, values_from = logFC, values_fill = NA) %>%
  tidyr::pivot_longer(cols = -Project, names_to = "gene_name", values_to = "logFC")
  
  

dat.wide <- dat.long %>%
  tidyr::pivot_wider(id_cols = gene_name, names_from = Project, values_from = logFC) %>%
  tibble::column_to_rownames("gene_name") %>%
  as.matrix()

dat.wide.num <- dat.long %>%
  dplyr::mutate(logFC = replace_na(logFC, 0)) %>%
  tidyr::pivot_wider(id_cols = gene_name, names_from = Project, values_from = logFC) %>%
  tibble::column_to_rownames("gene_name") %>%
  as.matrix()


# NOT RUN, check if the values all have onnly one gene for each project****
# All were good. 
#dat.wide <- dat.long %>%
#  dplyr::select(Project,gene_name, logFC) %>%
#  dplyr::group_by(gene_name, Project) %>%
#  dplyr::summarise(n = dplyr::n(), .groups = "drop") %>%
#  dplyr::filter(n > 1L)


coldata <- as.data.frame(colnames(dat.wide)) %>%
  dplyr::mutate("Projects" = "colnames(dat.wide)")

my_colors1 <- RColorBrewer::brewer.pal(8, "Dark2")
my_colors2 <- RColorBrewer::brewer.pal(8, "Pastel1")
colors <- c(my_colors1, my_colors2)
# First figure in the data visualization model.R is using the col.ll data 
my_colors_final <- c("BLCA"="#1B9E77", "BRCA"="#D95F02", "COAD"="#7570B3", "ESCA"="#E7298A", "HNSC"="#66A61E", "KICH"="#E6AB02", "KIRC"="#A6761D", "KIRP"="#666666", "LIHC"="#FBB4AE", "LUAD"="#B3CDE3", "LUSC"="#CCEBC5", "PRAD"="#DECBE4", "READ"="#FED9A6", "STAD"="#FFFFCC", "THCA"="#E5D8BD", "UCEC"="#FDDAEC")

proj <- colnames(dat.wide.num)

ha_column <- HeatmapAnnotation(
  "Projects" = c("BLCA","BRCA","COAD","ESCA","HNSC","KICH","KIRC","KIRP","LIHC","LUAD","LUSC","PRAD","READ","STAD","THCA","UCEC"),
  col = list("BLCA"="#1B9E77", "BRCA"="#D95F02", "COAD"="#7570B3", "ESCA"="#E7298A", "HNSC"="#66A61E", "KICH"="#E6AB02", "KIRC"="#A6761D", "KIRP"="#666666", "LIHC"="#FBB4AE", "LUAD"="#B3CDE3", "LUSC"="#CCEBC5", "PRAD"="#DECBE4", "READ"="#FED9A6", "STAD"="#FFFFCC", "THCA"="#E5D8BD", "UCEC"="#FDDAEC"))


# cluster the rows
row.dend <- as.dendrogram(hclust(dist(dat.wide.num)))
hm <- Heatmap(matrix = dat.wide.num,
  heatmap_legend_param = list(title = "Log2 Fold Change", at = c(-6, -3, 0,3,6)),
  top_annotation = ha_column,
  col = colorRamp2(c(-6, 0, 6), c("navy", "white", "red")),
  show_column_names = TRUE,
  column_names_gp = grid::gpar(fontsize = 9),
  show_row_names = FALSE,
  cluster_columns = FALSE,
  show_row_dend = TRUE,
  show_column_dend = FALSE,
  use_raster = TRUE)

row_ord.dat <- row_order(hm)

mat.ordered <- dat.wide[order(row_ord.dat),]
mat.ordered <- as.data.frame(mat.ordered)
mat.ordered$Genes <- rownames(mat.ordered)
data.table::fwrite(file = "/Volumes/My Passport/artBio_consultation/results/heatmaps/projects_fc.txt", x= mat.ordered, row.names = FALSE)

pdf("/Volumes/My Passport/artBio_consultation/results/heatmaps/projects_fc.pdf", width = 12, height = 6)
ComplexHeatmap::draw(hm,ht_gap = unit(1, "cm"),row_title = "Genes encoding surface proteins", column_title = "Differentially expressed genes encoding for surface proteins TCGA")
dev.off()


# Gene set enrichment analysis

hall.mark.pathway <- fgsea::gmtPathways(gmt.file = "/Volumes/My Passport/artBio_consultation/data/h.all.v2023.2.Hs.symbols.gmt")
kegg <- fgsea::gmtPathways(gmt.file = "/Volumes/My Passport/artBio_consultation/data/c2.cp.kegg_legacy.v2023.2.Hs.symbols.gmt")
immuno <- fgsea::gmtPathways(gmt.file = "/Volumes/My Passport/artBio_consultation/data/c7.all.v2023.2.Hs.symbols.gmt")
i <-  1
counter = 0
for(i in 1:length(projects)){
  proj <- projects[i]
  print(proj)
  
  res <- dat %>%
    dplyr::filter(Project == proj) %>%
    dplyr::arrange(desc(logFC))
  
  ranks <- res$logFC
  names(ranks) <- res$gene_name
  
  fgseaRes.hm <- as.data.frame(fgsea(hall.mark.pathway, ranks, minSize=15, maxSize = 500, nperm=1000)) %>%
    dplyr::mutate(Project = proj)
  
  up.regulated.hm <- fgseaRes.hm %>%
    dplyr::arrange(desc(NES)) %>%
    dplyr::slice(3) %>%
    dplyr::pull(pathway)
  
  down.regulated.hm <- fgseaRes.hm %>%
    dplyr::arrange(NES) %>%
    dplyr::slice(3) %>%
    dplyr::pull(pathway)
  
  top.paths.hm <- c(up.regulated.hm, down.regulated.hm)
  #----------------------------------------------#
  fgseaRes.kegg <- as.data.frame(fgsea(kegg, ranks, minSize=15, maxSize = 500, nperm=1000)) %>%
    dplyr::mutate(Project = proj)
  
  up.regulated.kegg <- fgseaRes.kegg %>%
    dplyr::arrange(desc(NES)) %>%
    dplyr::slice(3) %>%
    dplyr::pull(pathway)
  
  down.regulated.kegg <- fgseaRes.hm %>%
    dplyr::arrange(NES) %>%
    dplyr::slice(3) %>%
    dplyr::pull(pathway)
  
  top.paths.kegg <- c(up.regulated.kegg, down.regulated.kegg)
  #----------------------------------------------#
  fgseaRes.immuno <- as.data.frame(fgsea(immuno, ranks, minSize=15, maxSize = 500, nperm=1000)) %>%
    dplyr::mutate(Project = proj)
  
  up.regulated.immuno <- fgseaRes.immuno %>%
    dplyr::arrange(desc(NES)) %>%
    dplyr::slice(3) %>%
    dplyr::pull(pathway)
  
  down.regulated.immuno <- fgseaRes.immuno %>%
    dplyr::arrange(NES) %>%
    dplyr::slice(3) %>%
    dplyr::pull(pathway)
  
  top.paths.immuno <- c(up.regulated.immuno, down.regulated.immuno)
  
  if(counter == 0){
    temp.paths.hm <- top.paths.hm
    temp.paths.kegg <- top.paths.kegg
    temp.paths.immuno <- top.paths.immuno
    
    temp.hm <- fgseaRes.hm
    temp.kegg <- fgseaRes.kegg
    temp.immuno <- fgseaRes.immuno
    counter = counter +1
  } else{
    
    temp.paths.hm <- unique(c(temp.paths.hm, top.paths.hm))
    temp.paths.kegg <- unique(c(temp.paths.kegg, top.paths.kegg))
    temp.paths.immuno <- unique(c(temp.paths.kegg, top.paths.immuno))
    
    temp.hm <- rbind(temp.hm, fgseaRes.hm)
    temp.kegg <- rbind(temp.kegg, fgseaRes.kegg)
    temp.immuno <- rbind(temp.immuno, fgseaRes.immuno)
    
    
    
  }
    
}

data.table::fwrite(file = "/Volumes/My Passport/artBio_consultation/results/gsea/immuno_gsea.txt", x = temp.immuno)
data.table::fwrite(file = "/Volumes/My Passport/artBio_consultation/results/gsea/kegg_gsea.txt", x = temp.immuno)
data.table::fwrite(file = "/Volumes/My Passport/artBio_consultation/results/gsea/hallmark_gsea.txt", x = temp.immuno)
  

  
# 







```




```{r Run analysis, echo=FALSE, message=FALSE}

# main
tcga.projects <- c("ACC","BLCA","LGG","BRCA","CESC","ESCA","GBM","HNSC","KICH","KIRC","KIRP","LIHC","LUAD","LUSC","DLBC","MESO","OV","PAAD","PCPG","PRAD","READ","SARC","SKCM","STAD","TGCT","THYM","THCA","UCS","UCEC","UVM")
GESPs <- data.table::fread("/Volumes/My Passport/artBio_consultation/data/GESPs.txt")

for(i in 1:length(tcga.projects)){

  proj <- tcga.projects[i]
  print(proj)
  proj.res <- differential_expression_data_TCGA_projects(proj = proj)
  
  
  
  if(is.object(proj.res)){
    dataDEGs.labs <- label_results_table(proj.res, proj, GESPs)
    data.table::fwrite(file = str_glue("{proj}_de_gesps_table.txt"), x = dataDEGs.labs)
    
  }
  
}


list.of.DGE.res.tables <- list.files("/Volumes/My Passport/artBio_consultation/results/resTables/")
list.of.GESPs.res.tables <- list.files("/Volumes/My Passport/artBio_consultation/results/GESP_resTables/")

# iterate through the list of tables and pass them to the bar plot visualization functions. The functions will write out 
# projects specific analyses for the purposes of slide deck preparations

for(i in 1:length(list.of.DGE.res.tables)){
  val <- list.of.DGE.res.tables[i]
  val <- paste0("/Volumes/My Passport/artBio_consultation/results/resTables/", val)
  proj <- str_split(val, pattern = "_")
  proj <- proj[[1]][6]
  proj <- str_split(proj, pattern = "\\.")
  proj <- proj[[1]][1]
  p <- DGE.bar.plot(val = val)
  ggsave(filename = str_glue("/Volumes/My Passport/artBio_consultation/results/barplots/TCGA-{proj}_dge_counts.pdf"), plot = p, device = "pdf", width = 9, height = 7, units = "in")
  
  val <- list.of.GESPs.res.tables[i]
  val <- paste0("/Volumes/My Passport/artBio_consultation/results/GESP_resTables/", val)
  p <- GESP.bar.plot(val = val)
  ggsave(filename = str_glue("/Volumes/My Passport/artBio_consultation/results/barplots/TCGA-{proj}_gesp_dge_counts.pdf"), plot = p, device = "pdf", width = 9, height = 7, units = "in")
}

counter =0
for(i in 1:length(list.of.GESPs.res.tables)){
  val <- list.of.GESPs.res.tables[i]
  val <- paste0("/Volumes/My Passport/artBio_consultation/results/GESP_resTables/", val)
  res.table <- data.table::fread(val)
  
  temp.list <- gene.list.from.res.tables(res.table)
  
  
  if(counter == 0){
    gene.list <- temp.list
    counter = counter +1
  } else{
    gene.list <- c(gene.list, temp.list)
    gene.list <- unique(gene.list)
  }

}

de.gesps.list <- as.data.frame(gene.list)
colnames(de.gesps.list)[1] <- "DE_GESPs"
data.table::fwrite(file = "/Volumes/My Passport/artBio_consultation/results/DE_gesps.txt", x = de.gesps.list)





```



### Citations

- Colaprico, Antonio, et al. “TCGAbiolinks: an R/Bioconductor package for integrative analysis of TCGA data.” Nucleic acids research 44.8 (2015): e71-e71.

- Silva, Tiago C., et al. “TCGA Workflow: Analyze cancer genomics and epigenomics data using Bioconductor packages.” F1000Research 5 (2016). (https://f1000research.com/articles/5-1542/v2)

- Mounir, Mohamed, et al. “New functionalities in the TCGAbiolinks package for the study and integration of cancer data from GDC and GTEx.” PLoS computational biology 15.3 (2019): e1006701. (https://doi.org/10.1371/journal.pcbi.1006701)









